feat: add `blas/ext/find-last-index`

headlessNode opened the pull request:
Progresses #2656.

## Description

> What is the purpose of this pull request?

This pull request:

-   add `blas/ext/find-last-index`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   progresses #2656

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

headlessNode committed:
feat: add blas/ext/find-last-index

headlessNode commented:
Need to merge https://github.com/stdlib-js/stdlib/pull/7786 first.

kgryte commented:
/stdlib merge

kgryte commented:
@headlessNode Mind doing a run through and comparing to `blas/ext/find-index` to ensure no changes need to be made to this PR before I review?

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into blas-ext-findLastIndex

headlessNode committed:
refactor: apply review suggestions

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte committed:
Discard changes to lib/node_modules/@stdlib/blas/ext/find-index/docs/repl.txt

kgryte commented on lib/node_modules/@stdlib/blas/ext/find-last-index/docs/types/index.d.ts:
```diff
@@ -0,0 +1,276 @@
+}
+
+/**
+* Interface describing `findLastIndex`
```
```suggestion
* Interface describing `findLastIndex`.
```

kgryte committed:
docs: add missing period

Signed-off-by: Athan <kgryte@gmail.com>

kgryte committed:
docs: add missing notes

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/ext/find-last-index/docs/types/test.ts:
```diff
@@ -0,0 +1,405 @@
+	findLastIndex( x, { 'dim': ( x: number ): number => x }, clbk ); // $ExpectError
+
+	findLastIndex( x, { 'dim': '5' }, clbk, {} ); // $ExpectError
+	findLastIndex( x, { 'dim': 5 }, clbk, {} ); // $ExpectError
```
```suggestion
```

kgryte committed:
test: remove invalid test case

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/ext/find-last-index/lib/assign.js:
```diff
@@ -0,0 +1,167 @@
+		throw new TypeError( format( 'invalid argument. The first argument must be an ndarray. Value: `%s`.', x ) );
+	}
+	if ( !isndarrayLike( out ) ) {
+		throw new TypeError( format( 'invalid argument. The second argument must be an ndarray. Value: `%s`.', out ) );
```
```suggestion
		throw new TypeError( format( 'invalid argument. First argument must be an ndarray. Value: `%s`.', x ) );
	}
	if ( !isndarrayLike( out ) ) {
		throw new TypeError( format( 'invalid argument. Second argument must be an ndarray. Value: `%s`.', out ) );
```

kgryte committed:
fix: update error messages

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/ext/find-last-index/lib/main.js:
```diff
@@ -0,0 +1,161 @@
+			opts.dtype = options.dtype;
+		}
+	}
+	// Resolve the list of non-reduced dimensions:
```
```suggestion
```

kgryte committed:
docs: remove obsolete comment

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (approved):
LGTM
